pkg/apis/auditregistration/types.go

```diff
@@ -104,11 +102,37 @@ type Policy struct {
 	Level Level
 
 	// Stages is a list of stages for which events are created.
+	// Must be non-empty if level != none
+	// +optional
+	Stages []Stage
+
+	// Rules define how classes should be handled.
+	// A request may fall under multiple audit classes.
+	// Rules are evaluated in order (first matching wins).
+	// Rules override the top-level & stage.
+	// Unmatched requests use the top-level rule.
+	// +optional
+	Rules []PolicyRule
+}
+
+// PolicyRule defines how a class is handled per sink
+type PolicyRule struct {
+	// ClassName of the AuditClass object. This rule matches requests that are
+	// classified with this AuditClass
+	ClassName string
+
+	// The Level that all requests are recorded at.
+	// available options: None, Metadata, Request, RequestResponse
+	// required
+	Level Level
+
+	// Stages is a list of stages for which events are created.
+	// If no stages are given nothing will be logged
```

Comment:
The policy-level `Stages` member also requires that `Must be non-empty if level != none`. This one doesn't explicitly require such constraints. Care to explain the difference in the approach?